lib/api/src/rest/validate.rs

```diff
@@ -33,14 +33,14 @@ impl Validate for QueryInterface {
 impl Validate for Query {
     fn validate(&self) -> Result<(), validator::ValidationErrors> {
         match self {
-            Query::Nearest(vector) => vector.nearest.validate(),
-            Query::Recommend(recommend) => recommend.recommend.validate(),
-            Query::Discover(discover) => discover.discover.validate(),
-            Query::Context(context) => context.context.validate(),
-            Query::Fusion(fusion) => fusion.fusion.validate(),
+            Query::Nearest(vector) => vector.validate(),
```

Comment:
those lines make sure to use the validation at the struct level to make it fool proof when adding new fields